cargo-shuttle/src/lib.rs

```diff
@@ -427,21 +437,150 @@ impl Shuttle {
             args.path.clone()
         };
 
-        // 4. Ask for the framework
+        // 4. Ask for the template
         let template = match git_template {
             Some(git_template) => git_template,
             None => {
-                println!(
-                    "Shuttle works with a range of web frameworks. Which one do you want to use?"
-                );
-                let frameworks = args::InitTemplateArg::iter().collect::<Vec<_>>();
-                let index = FuzzySelect::with_theme(&theme)
-                    .with_prompt("Framework")
-                    .items(&frameworks)
-                    .default(0)
-                    .interact()?;
-                println!();
-                frameworks[index].template()
+                // /// Can be used during testing
+                // async fn get_schema() -> Result<TemplatesSchema> {
+                //     Ok(toml::from_str(include_str!(
+                //         "../../examples/templates.toml"
+                //     ))?)
+                // }
+                async fn get_schema() -> Result<TemplatesSchema> {
+                    let client = reqwest::Client::new();
+                    Ok(toml::from_str(
+                        &client
+                            .get(shuttle_common::constants::EXAMPLES_TEMPLATES_TOML)
+                            .send()
+                            .await?
+                            .text()
+                            .await?,
+                    )?)
+                }
```

Comment:
It looks like this is just used once, can't we call it directly rather than having an inner function?

Reply:
Having it like this makes it easier to hot-swap to another impl (see other comment)

Reply:
It also groups the `?` error returns to be easily handled as one in `map_err`

Reply:
I don't want to block the release, but it feels quite messy to me. Can we rather have a comment inside/over the function that you can use a template from the examples directly for testing? This function could also be external, even though it's only used here, since this parent function is very large already.